datanode/dehistory/snapshot/service_create_snapshot.go

```diff
@@ -84,26 +76,18 @@ func (b *Service) createSnapshot(ctx context.Context, chainID string, fromHeight
 		runAllInReverseOrder(cleanUp)
 		return CreateSnapshotResult{}, fmt.Errorf("failed to create write lock file:%w", err)
 	}
-	cleanUp = append(cleanUp, func() { os.Remove(snapshotInProgressFile) })
-
-	if async {
-		go func() {
-			defer func() { runAllInReverseOrder(cleanUp) }()
-			err = b.snapshotData(ctx, copyDataTx, dbMetaData, currentSnapshot, historySnapshot)
-			if err != nil {
-				b.log.Error("failed to snapshot data", logging.Error(err))
-				if b.config.PanicOnSnapshotCreationError {
-					panic(fmt.Sprintf("failed to snapshot data:%s", err))
-				}
-			}
-		}()
-	} else {
+	cleanUp = append(cleanUp, func() { _ = os.Remove(snapshotInProgressFile) })
+
+	go func() {
 		defer func() { runAllInReverseOrder(cleanUp) }()
 		err = b.snapshotData(ctx, copyDataTx, dbMetaData, currentSnapshot, historySnapshot)
 		if err != nil {
-			return CreateSnapshotResult{}, fmt.Errorf("failed to snapshot data:%w", err)
+			b.log.Error("failed to snapshot data", logging.Error(err))
+			if b.config.PanicOnSnapshotCreationError {
+				panic(fmt.Sprintf("failed to snapshot data:%s", err))
```

Comment:
why not use b.log.Panic() here instead?

Reply:
it did originally, but got an error using that and it did not log the original error, so I change it to this instead so that we always get the cause